src/mpid/ch4/src/ch4r_callbacks.c

```diff
@@ -6,6 +6,7 @@
 #include "mpidimpl.h"
 #include "mpidch4r.h"
 #include "ch4r_callbacks.h"
+#include "mpidig_am_gpu.h"
```

Comment:
All the `ch4r` functions will need be moved to `generic/am/`, right? So let's not add more legacy code and start to the right thing. How about move these code to `generic/am/mpidig_am_gpu.c`? The init function -- the one registers the call back need be in `mpidig_am_gpu.c` as well.